src/node/consumer/src/lib/models/composition_root.ts

```diff
@@ -0,0 +1,27 @@
+import {ICompositionalRoot} from "@interfaces/composition_root";
+import {IProcessTorrentsJob} from "@interfaces/process_torrents_job";
+import {ITrackerService} from "@interfaces/tracker_service";
+import {IocTypes} from "@models/ioc_types";
+import {IDatabaseRepository} from "@repository/interfaces/database_repository";
+import {inject, injectable} from "inversify";
+
+@injectable()
+export class CompositionalRoot implements ICompositionalRoot {
+    private trackerService: ITrackerService;
+    private databaseRepository: IDatabaseRepository;
+    private processTorrentsJob: IProcessTorrentsJob;
+
+    constructor(@inject(IocTypes.ITrackerService) trackerService: ITrackerService,
+                @inject(IocTypes.IDatabaseRepository) databaseRepository: IDatabaseRepository,
+                @inject(IocTypes.IProcessTorrentsJob) processTorrentsJob: IProcessTorrentsJob) {
+        this.trackerService = trackerService;
+        this.databaseRepository = databaseRepository;
+        this.processTorrentsJob = processTorrentsJob;
+    }
```

Comment:
With inversify we can go one step further and remove the constructors

```js
@injectable()
export class TorrentFileService implements ITorrentFileService {
    @inject(IocTypes.IMetadataService) metadataService: IMetadataService;
    @inject(IocTypes.ITorrentDownloadService) torrentDownloadService: ITorrentDownloadService;
    @inject(IocTypes.ILoggingService) logger: ILoggingService;
```

I've done this and updated all the tests to resolve objects from the ioc container instead of through the constructor

Reply:
Oh that's way nicer! Thanks.